tracing-core/src/metadata.rs

```diff
@@ -365,10 +365,13 @@ impl From<Level> for LevelFilter {
     }
 }
 
-impl Into<Option<Level>> for LevelFilter {
+impl From<Option<Level>> for LevelFilter {
     #[inline]
-    fn into(self) -> Option<Level> {
-        self.into_level()
+    fn from(level: Option<Level>) -> Self {
+        match level {
+            Some(level) => LevelFilter::from_level(level),
+            None => LevelFilter::OFF,
+        }
```

Comment:
nit, TIOLI: can't this just be
```suggestion
        Self(level)
```